src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.cs

```diff
@@ -6,9 +6,27 @@ namespace System.Globalization
 {
     internal static partial class GlobalizationMode
     {
+        internal static bool UseIcu { get; private set; } = GetUseIcuMode();
+
         private static bool GetGlobalizationInvariantMode()
         {
             return GetInvariantSwitchValue();
         }
+
+        private static bool GetUseIcuMode()
+        {
+            bool useNlsValue = GetUseNlsSwitchValue();
```

Comment:
Why is this before the `Invariant` check?

Reply:
It was to return !useNlsValue... but I don't think it makes sense, if Invariant is set, it shouldn't matter what the value of `UseIcu` is.

Reply:
yes I have the same question as Jan.

---
In reply to: [405038481](https://github.com/dotnet/runtime/pull/34645#discussion_r405038481) [](ancestors = 405038481)